### ISSUE TITLE
Enable tab to search from browser address bar

### ISSUE THREAD

**Issue by IgorMinar** [angular/angular#24107]
[Most modern browsers](http://www.opensearch.org/Community/OpenSearch_search_clients) support a spec called [opensearch](https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md). This spec enables the following usage pattern:

- open browser
- tab into the address bar
- type a domain (or part of the domain)
- hit tab
- type search query
- hit enter
- the specified domain opens at the search result page for the given query

In more advanced use-cases the top search results could even appear in the address bar autocompletion list, but for now let's not worry about that and focus only on the flow above. 

This is what the experience look like in chrome for medium.com when a user types the query:

![screenshot 2018-05-24 at 8 31 09 am](https://user-images.githubusercontent.com/216296/40496092-c0d937ee-5f2d-11e8-80ec-1fc63d0dee2f.png)

And this is what the result looks like after the user hits enter:

![screenshot 2018-05-24 at 8 37 17 am](https://user-images.githubusercontent.com/216296/40496112-cb36099c-5f2d-11e8-9fe0-a50b7a6b9f49.png)

This feature is relatively easy to implement for angular.io. There are two main pieces to it:
- ensure that we can activate the search via a url
  - by navigating to angular.io/?q=<query> we should start the app and open up the search results for query `<query>`
  - we might consider different url scheme for example that used by auxiliary routes in the Angular router or create a dedicated `/search` page - this needs to be thought through
  - if we go with non-dedicated page and we just overlay the search results over e.g. the homepage, we need to ensure that we clean up the url once the search result overlay is closed
  - we also need to think a bit about how to report these hits in analytics, also see #23581 
- add open search descriptor to the index.html. I found the following resources helpful to understand how this should be done:
  - http://www.chromium.org/tab-to-search
  - http://blog.coredumped.org/2012/04/opensearch-how-to-get-google-chrome-to.html

**Comment by s-gbz** [comment]
I'll get down on it 👍

**Comment by sarunint** [comment]
Related to #19462

**Comment by s-gbz** [comment]
**Update:**
Writing the open_search.xml file was easy. The struggle I just faced was testing,
because neither Chrome nor Firefox are able to properly download and hence integrate the file.

E.g. 
```
  <link rel="icon" type="image/x-icon" href="favicon.ico">
```
opens the icon.
```
  <link rel="search" type="application/opensearchdescription+xml" href="osd.xml" download/>
```
results in 404.

Right clicking the link to "Save as", results in 

> Error: File not found

I don't know if it's related to the localhost environment or some Angular specific issue. I'll keep trying.

Did some face a similiar problem before?

**Comment by gkalpak** [comment]
I am not sure what you mean :grin:
Feel free to create a WIP pull request with what you have right now and describe the issue you are having (i.e. what you are trying to accomplish and what happens instead). It will be much easier to offer guidance.

Thx for working on this btw ❤️

**Comment by s-gbz** [comment]
Hey guys, just wanted to let you know, that I had no spare time to work this out, but be back on it after holidays in 3 weeks :)

**Comment by sarunint** [comment]
Closed via #25479

**Comment by gkalpak** [comment]
For some reason, it works for me for next.angular.io, but not for angular.io 😕
